### PR TITLE
fix 'Wrong processing result for `div.innerHtml=<p><header></header>`' (close #688)

### LINKED ISSUE #688: Wrong processing result for `div.innerHtml=<p><header></header>`
Without hammerhead:
`div.innerHTML = "<p><header></header>"`;
`div.innerHTML`
`<p></p><header></header>`

With hammerhead:
`div.innerHTML = "<p><header></header>"`;
`div.innerHTML`
`<p><header></header></p>`

Test code from http://html5test.com
Problem inside `htmlUtils.processHtml`

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev @VasilyStrelyaev

### CODE REVIEW

File: test/client/fixtures/utils/html-test.js [hunk 1]
```diff
@@ -272,15 +272,19 @@ test('html and body attributes must be processed (T226655)', function () {
...
         '<script>',
-        processScript('\n    var x = y <this.length, y = 3 > 5;\n', true),
+        processScript('\n    var x = y <th .length, y = 3 > 5;\n', true),
         '<\/script>'
     ].join('');
 
     strictEqual(htmlUtils.processHtml(htmlSrc), htmlExpected);
 });
+
+test('process html with non-closed the "p" tag and the "header" tag (GH-688)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/utils/html-test.js]:
```
test('process html with an unclosed "p" tag and the "header" tag (GH-688)', function () {
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by LavrovArtem** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by miherlosev** [comment]
lgtm

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/utils/html-test.js [hunk 2]
```diff
@@ -272,15 +272,25 @@ test('html and body attributes must be processed (T226655)', function () {
 test('html with special script is processed correctly (GH-684)', function () {
     var htmlSrc = [
         '<script>',
-        '    var x = y <this.length, y = 3 > 5;',
+        // We need some kind of table tags in js code e.g. <th .length, y = 3 >
```
Comment (churkin) [inline test/client/fixtures/utils/html-test.js]: NOTE:

**Comment by churkin** [comment]
-\r

**Comment by LavrovArtem** [comment]
ping @churkin

**Comment by churkin** [comment]
lgtm

**Pull request merged**